frame/nfts/src/lib.rs

```diff
@@ -1768,6 +1789,35 @@ pub mod pallet {
 				witness_price,
 			)
 		}
+
+		/// Mint an item by providing the pre-signed approval.
+		///
+		/// Origin must be Signed.
+		///
+		/// - `data`: The pre-signed approval that consists of the information about the item, its
+		///   metadata, attributes, who can mint it (`None` for anyone) and until what block number.
+		/// - `signature`: The signature of the `data` object.
+		/// - `signer`: The `data` object's signer. Should be an owner of the collection.
+		///
+		/// Emits `Issued` on success.
+		/// Emits `ItemMetadataSet` if the metadata was not empty.
+		#[pallet::call_index(37)]
+		#[pallet::weight(T::WeightInfo::mint_pre_signed(data.attributes.len() as u32))]
+		pub fn mint_pre_signed(
+			origin: OriginFor<T>,
+			data: PreSignedMintOf<T, I>,
+			signature: MultiSignature,
+			signer: MultiSigner,
```

Comment:
So we are forcing the off-chain users to use the same crypto as on-chain.  
I will just locally give it a shot to make it generic.

Reply:
I changed it here locally to be generic https://github.com/paritytech/substrate/compare/js/offchain-mint...oty-offchain-sig

Reply:
Thanks, I moved all the suggested changes from your branch into that PR